explore/src/main/scala/explore/observationtree/ObsList.scala

```diff
@@ -60,121 +55,112 @@ object ObsList {
       ObsSummaryWithTargetsAndConstraints.id
     )
 
-  protected class Backend {
-    protected def insertObs(
-      pos:        Int,
-      focusedObs: ReuseView[Option[Observation.Id]],
-      undoCtx:    UndoContext[ObservationList]
-    )(implicit
-      c:          TransactionalClient[IO, ObservationDB]
-    ): Callback = {
-      // Temporary measure until we have id pools.
-      val newObs = CallbackTo(Random.nextInt(0xfff)).map(int =>
-        ObsSummaryWithTargetsAndConstraints(
-          Observation.Id(PosLong.unsafeFrom(int.abs.toLong + 1)),
-          targets = List.empty,
-          constraints = ConstraintsSummary.default,
-          ObsStatus.New,
-          ObsActiveStatus.Active,
-          Duration.ZERO
-        )
-      )
+  protected def insertObs(
+    pos:        Int,
+    focusedObs: ReuseView[Option[Observation.Id]],
+    undoCtx:    UndoContext[ObservationList],
+    adding:     ReuseView[Boolean]
+  )(implicit
+    ctx:        AppContextIO
+  ): IO[Unit] =
+    adding.set(true).to[IO] >>
```

Comment:
We can also use `adding.async.set` here.

Reply:
That seems a little cleaner. I'll change that in a few places.